2048.h

```diff
@@ -28,7 +28,8 @@ static inline void print_board(board_t board) {
     int i,j;
     for(i=0; i<4; i++) {
         for(j=0; j<4; j++) {
-            printf("%c", "0123456789abcdef"[(board)&0xf]);
+            uint8_t powerVal = (board) & 0xf;
+            printf("%*u ", 6, powerVal == 0 ? 0 : 1 << powerVal);
```

Comment:
Sorry, one more nitpick: I think it looks odd to use `%*u` with a constant width. Maybe just `%6u`? Then you also don't need the trailing space since we can't represent tiles > 32768 anyway. Otherwise looks good, and I'll merge.